data-prepper-plugins/kafka-connect-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafkaconnect/util/KafkaConnect.java

```diff
@@ -0,0 +1,370 @@
+        }
+    }
+
+    /**
+     * Stop the Kafka Connect.
+     */
+    public void stop() {
+        if (this.connect == null) {
+            LOG.info("Kafka Connect is running, will not start again");
+            return;
+        }
+        LOG.info("Stopping Kafka Connect");
+        this.connect.stop();
+    }
+
+    private DistributedHerder initHerder(Map<String, String> workerProps, DistributedConfig config, RestClient restClient) {
+        LOG.info("Scanning for plugin classes. This might take a moment ...");
+        Plugins plugins = new Plugins(workerProps);
+        plugins.compareAndSwapWithDelegatingLoader();
+        String kafkaClusterId = config.kafkaClusterId();
+        LOG.info("Kafka cluster ID: {}", kafkaClusterId);
+
+        URI advertisedUrl = rest.advertisedUrl();
+        String workerId = advertisedUrl.getHost() + ":" + advertisedUrl.getPort();
+
+        String clientIdBase = ConnectUtils.clientIdBase(config);
+
+        // Create the admin client to be shared by all backing stores.
+        Map<String, Object> adminProps = new HashMap<>(config.originals());
+        ConnectUtils.addMetricsContextProperties(adminProps, config, kafkaClusterId);
+        adminProps.put(CLIENT_ID_CONFIG, clientIdBase + "shared-admin");
+        SharedTopicAdmin sharedAdmin = new SharedTopicAdmin(adminProps);
+
+        KafkaOffsetBackingStore offsetBackingStore = new KafkaOffsetBackingStore(sharedAdmin, () -> clientIdBase,
+                plugins.newInternalConverter(true, JsonConverter.class.getName(),
+                        Collections.singletonMap(JsonConverterConfig.SCHEMAS_ENABLE_CONFIG, "false")));
+        offsetBackingStore.configure(config);
+
+        ConnectorClientConfigOverridePolicy connectorClientConfigOverridePolicy = plugins.newPlugin(
+                config.getString(WorkerConfig.CONNECTOR_CLIENT_POLICY_CLASS_CONFIG),
+                config, ConnectorClientConfigOverridePolicy.class);
+
+        Worker worker = new Worker(workerId, time, plugins, config, offsetBackingStore, connectorClientConfigOverridePolicy);
+        WorkerConfigTransformer configTransformer = worker.configTransformer();
+
+        Converter internalValueConverter = worker.getInternalValueConverter();
+        StatusBackingStore statusBackingStore = new KafkaStatusBackingStore(time, internalValueConverter, sharedAdmin, clientIdBase);
+        statusBackingStore.configure(config);
+
+        ConfigBackingStore configBackingStore = new KafkaConfigBackingStore(
+                internalValueConverter,
+                config,
+                configTransformer,
+                sharedAdmin,
+                clientIdBase);
+
+        // Pass the shared admin to the distributed herder as an additional AutoCloseable object that should be closed when the
+        // herder is stopped. This is easier than having to track and own the lifecycle ourselves.
+        return new DistributedHerder(config, time, worker,
```

Comment:
Is there any metrics that we should monitor for this class ?

Reply:
This class is owned by KafkaConnect Runtime. We cannot pass-in metrics object.
Here is the things we can monitor: https://docs.confluent.io/platform/current/connect/monitoring.html
As mentioned in our low-level design review, we can keep add those as needed.